content/blog/2021/04/2021-04-20-tekton-plugin.adoc

```diff
@@ -0,0 +1,178 @@
+
+== Comparing the Kubernetes and Tekton plugins
+
+Those of you using Jenkins on a link:https://kubernetes.io/[Kubernetes] cluster are probably using the link:https://plugins.jenkins.io/kubernetes/[kubernetes plugin] right now.
+
+link:https://github.com/jenkinsci/kubernetes-plugin/tree/master/examples/declarative_from_yaml_file[Here is an example] of how to use a link:https://github.com/jenkinsci/kubernetes-plugin/blob/master/examples/declarative_from_yaml_file/Jenkinsfile[Jenkinsfile] with a link:https://github.com/jenkinsci/kubernetes-plugin/blob/master/examples/declarative_from_yaml_file/KubernetesPod.yaml[pod YAML file] so that you can run commands in different containers in the pod.
+
+What this means is that:
+
+* a kubernetes pod is created based on the link:https://github.com/jenkinsci/kubernetes-plugin/blob/master/examples/declarative_from_yaml_file/KubernetesPod.yaml[pod YAML file] which is scheduled by kubernetes
+* the link:https://github.com/jenkinsci/kubernetes-plugin/blob/master/examples/declarative_from_yaml_file/Jenkinsfile[Jenkinsfile] runs on the Jenkins controller talking over Jenkins remoting to the pod to tell it to run commands in different containers. The pod includes the `jnlp` container which does the remoting between the Jenkins controller and the pod
+
+This has a few issues:
+
+* each container in the pod must have a shell so that jnlp can invoke commands. This may mean you have to create your own images
+* it can be a little slow to start since there is chattiness with the Jenkins controller and the pod - whereas with Tekton pods just start and run locally without any coodination with the Jenkins controller
+* you have to maintain 2 files: the `Jenkinsfile` and the `pod.yaml` and it's hard to share/override both of those files across multiple repositories as you need to make changes (e.g. overriding environment variables/images/commands/resource limits on demand on steps).
+
+Though one downside of the tekton approach is that by default there is no automatic synchronisation of state; after a Task in tekton completes there's no automatic upload of state to the Jenkins controllers disk. You can always add a step in your Task to upload workspace state to the Jenkins controller if that's what you want.
+
+Though remember that tekton plugin doesn't take anything away; so you can mix and match the kubernetes and tekton plugins to suit your needs.
+
+
+== Conclusion
+
+We are really excited about the combination of Jenkins, link:https://tekton.dev/[Tekton] and link:https://jenkins-x.io/[Jenkins X] letting developers pick the best tool for the job while becoming more cloud native and increasing the automation help reduce the amount of manual work creating and maintaining pipelines while also helping to improve the quality and practices of our CI/CD.
+
+Please try it out and link:https://github.com/jenkinsci/tekton-client-plugin/issues[let us know how you get on]
```

Comment:
Add a link to https://github.com/jenkinsci/tekton-client-plugin/blob/master/CONTRIBUTING.md as well?